src/cartridges/int_adyen_overlay/cartridge/scripts/checkNotificationAuth.js

```diff
@@ -40,6 +41,30 @@ function check(request) {
   );
 }
 
+function compareHmac(hmacSignature, merchantSignature){
+  let bitwiseComparison;
+  if (hmacSignature.length !== merchantSignature.length){
+    return false;
+  }
+  for (let i = 0; i < hmacSignature.length; i++) {
+    bitwiseComparison |= hmacSignature.charCodeAt(i) ^ merchantSignature.charCodeAt(i);
+  }
+  return bitwiseComparison === 0;
+};
+
+function validateHmacSignature(request){
+  const notificationData = request.form;
+  const hmacSignature = notificationData['additionalData.hmacSignature'];
+  const merchantSignature = AuthenticationUtils.calculateHmacSignature(request);
+  // Checking for timing attacks
+  if (compareHmac(hmacSignature, merchantSignature)){
+    AdyenLogs.info_log(`Successfully compared the HMAC Signatures, notification is coming from Adyen`);
```

Comment:
I don't think there is a need to log anything when it's successful. Instead we can add a log when it errors to say that there is a mismatch

Reply:
Updated it to the error scenario